src/lib/cookieBuilder.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright 2021 Curity AB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import {CookieSerializeOptions, serialize} from 'cookie'
+import {getEncryptedCookie} from './cookieEncrypter'
+import OAuthAgentConfiguration from './oauthAgentConfiguration'
+import {getATCookieName, getAuthCookieName, getCSRFCookieName, getIDCookieName} from './cookieName'
+import {getTempLoginDataCookieForUnset} from './pkce'
+
+function getCookiesForFailedLoginResponse(config: OAuthAgentConfiguration): string[] {
+
+    return [
+        getTempLoginDataCookieForUnset(config.cookieOptions, config.cookieNamePrefix)
+    ]
+}
+
+function getCookiesForTokenResponse(tokenResponse: any, config: OAuthAgentConfiguration, unsetTempLoginDataCookie: boolean = false, csrfCookieValue?: string): string[] {
+    
+    const cookies = [
+        getEncryptedCookie(config.cookieOptions, tokenResponse.access_token, getATCookieName(config.cookieNamePrefix), config.encKey)
+    ]
+
+    if (csrfCookieValue) {
+        cookies.push(getEncryptedCookie(config.cookieOptions, csrfCookieValue, getCSRFCookieName(config.cookieNamePrefix), config.encKey))
+    }
+
+    if (unsetTempLoginDataCookie) {
+        cookies.push(getTempLoginDataCookieForUnset(config.cookieOptions, config.cookieNamePrefix))
+    }
+
+    if (tokenResponse.refresh_token) {
+        const refreshTokenCookieOptions = {
+            ...config.cookieOptions,
+            path: config.endpointsPrefix + '/refresh'
+        }
+        cookies.push(getEncryptedCookie(refreshTokenCookieOptions, tokenResponse.refresh_token, getAuthCookieName(config.cookieNamePrefix), config.encKey))
+    }
+
+    if (tokenResponse.id_token) {
+        const idTokenCookieOptions = {
```

Comment:
Nice to have - this could be calculated once somewhere, but I don't have an idea now how to do it elegantly.